colorbleed/plugins/fusion/inventory/select_loader_color.py

```diff
@@ -0,0 +1,40 @@
+from avalon import api
+from avalon.vendor.Qt import QtGui, QtWidgets
+
+import avalon.fusion
+
+
+class FusionSelectLoaderColor(api.InventoryAction):
+    """Update the color of the selected tools"""
+
+    label = "Select Loader Color"
+    icon = "plus"
+    color = "#d8d8d8"
+
+    def process(self, containers):
+        """Color all selected tools the selected colors"""
+
+        comp = avalon.fusion.get_current_comp()
+
+        # Launch pick color
+        color = QtGui.QColor(1.0, 1.0, 1.0)
+        picked_color = QtWidgets.QColorDialog().getColor(color)
+        with avalon.fusion.comp_lock_and_undo_chunk(comp):
+            for container in containers:
+                # Convert color to RGB 0-1 floats
+                rgb_f = picked_color.getRgbF()
+                rgb_f_table = {"R": rgb_f[0], "G": rgb_f[1], "B": rgb_f[2]}
+
+                # Update tool
+                tool = container["_tool"]
+                tool.TileColor = rgb_f_table
+
+        self.refresh()
+
+    def refresh(self):
```

Comment:
Ah, interesting! Actually... looking at this. Whenever an action is perform in the Scene Inventory, shouldn't it always update afterwards right in the Scene Inventory? Let's see if we can remove this here - otherwise I'm expecting we'll start having hacks like this in many actions.

Reply:
Enabling actions to trigger the host's `refresh`  in a non-hack way would be very nice to have :+1: